### PR TITLE
Deadlock resolution for MOFED 5.4-3.0.0.0 and NV Peer memory 1.2

### PR DESCRIPTION
**Pull request by abhamidipati0614**
Minor changes:

-  dkms 2.8.6-1.el7 update
-  utility azcopy
- WALinuxAgent update v2.5.0.1 for ubuntu images
- IMPI_2021 update v2021.4.0 
- gcc tarball from azhpcstor

### CODE REVIEW

**Review by jithinjosepkl — APPROVED** [review]

File: ubuntu/common/hpc-tuning.sh [hunk 1]
```diff
@@ -29,20 +29,22 @@ sysctl -p
...
+wget ${DOWNLOAD_URL}
+tar -xvf $(basename ${DOWNLOAD_URL})
+pushd WALinuxAgent-pre-${WAAGENT_VERSION}/
 python3 setup.py install --register-service
 popd
 
 # Configure WALinuxAgent
 sed -i -e 's/# OS.EnableRDMA=y/OS.EnableRDMA=y/g' /etc/waagent.conf
-echo "Extensions.GoalStatePeriod=120" | sudo tee -a /etc/waagent.conf
+echo "Extensions.GoalStatePeriod=300" | sudo tee -a /etc/waagent.conf
```
Comment (jithinjosepkl) [inline ubuntu/common/hpc-tuning.sh:43]: Dont we need this for CentOS also?

Comment (jithinjosepkl) [reply ubuntu/common/hpc-tuning.sh:43]: @mvrequa  - please review the waagent config changes.

File: centos/centos-8.x/common/install_utils.sh [hunk 2]
```diff
@@ -30,4 +30,24 @@ yum install -y numactl \
     libpciaccess \
     cmake \
     libnl3-devel \
-    libarchive
+    libarchive \
+    libsecret \
+    epel-release
```
[the remark this answers is not in the document]
Comment (abhamidipati0614) [reply centos/centos-8.x/common/install_utils.sh:35]: the package htop is dependent on epel-release

File: ubuntu/common/hpc-tuning.sh (hunk 1 above)
Comment (abhamidipati0614) [reply ubuntu/common/hpc-tuning.sh:43]: > Dont we need this for CentOS also?

Added

**Review by jithinjosepkl — APPROVED** [review]

**Review by jithinjosepkl — APPROVED** [review]

File: ubuntu/common/hpc-tuning.sh (hunk 1 above)
Comment (mvrequa) [reply ubuntu/common/hpc-tuning.sh:43]: May we add this? If not set then GoalStatePeriod is used for the initial converge.
 Extensions.InitialGoalStatePeriod=6

**Pull request merged**